frontend/src/certificate_verification.tsx

```diff
@@ -0,0 +1,266 @@
+
+  const verifyCertificate = (id: string) => {
+    const postCertificate = (
+      id: string,
+      content: string | ArrayBuffer | undefined
+    ) => {
+      fetch(`/api/certificates/verify/${id}`, {
+        body: content,
+        method: "POST",
+        credentials: "same-origin",
+        headers: {
+          "Content-Type": "application/pdf",
+        },
+      })
+        .then((res) => {
+          if (res.status === 200) {
+            setCertValid(true);
+          } else if (res.status === 202) {
+            setCertOutdated(true);
+          } else if (res.status === 404) {
+            setCertNotFound(true);
+          } else {
+            throw new Error("Något gick fel.");
+          }
+        })
+        .catch((error) => {
+          userMessage(error.message, "error");
+        });
+    };
+
+    if (file) {
+      const reader = new FileReader();
+      reader.onload = (event) => {
+        const binaryFile = event.target?.result
+          ? event.target.result
+          : undefined;
+        postCertificate(id, binaryFile);
+      };
+      reader.readAsArrayBuffer(file);
+    }
+  };
+
+  const getLatestCertificate = (id: string) => {
+    fetch(`/api/certificates/preview/${id}`, {
+      method: "GET",
+      credentials: "same-origin",
+      headers: {
+        Accept: "application/octet-stream",
+        "Content-Type": "application/json",
+      },
+    })
+      .then((res) => res.arrayBuffer())
+      .then((data) => {
+        setPreviewWidth(window.innerWidth * 0.5);
+        setPreviewUrl(data);
+      })
+      .catch((error) => {
+        userMessage(error.message, "error");
+      });
+  };
+
+  // This function is necessary to make the previw work.
+  // Built into the library.
+  function onDocumentLoadSuccess({ numPages }) {
+    setNumPages(numPages);
+  }
+
+  return (
+    <>
+      <h1>Verifiera certifikat</h1>
+      <p>
+        Ladda upp ett digitalt certifikat för att se att det är signerat och
+        finns i databasen.
+      </p>
+      <div className={style.fileInput}>
+        <input
+          id="inputFile"
+          type="file"
+          onChange={(event) =>
+            setFile(event.target.files ? event.target.files[0] : undefined)
+          }
+          style={{ display: "none" }}
+        />
+        <label htmlFor="inputFile" className={style.label}>
+          <Button variant="contained" color="primary" component="span">
+            Välj fil...
+          </Button>
+        </label>
+        {file ? <p>{file.name}</p> : <p>Ingen fil vald.</p>}
+      </div>
+      <div className={style.flexRow}>
+        <Button
+          variant="contained"
+          color="primary"
+          onClick={() => {
+            setCertValid(false);
+            setCertOutdated(false);
+            setCertNotFound(false);
+            setShowPdf(false);
+            verifyCertificate(id);
+          }}
+          disabled={file ? false : true}
+        >
```

Comment:
What about allowing the user deleting the file before it's sent to verify? is this important?

Reply:
This shouldn't be different from other failures to read the file, as handled in the `onload` event.